alacritty/src/display/mod.rs

```diff
@@ -484,12 +484,17 @@ impl Display {
         config: &Config,
         search_state: &SearchState,
     ) {
+        let mut grid_cells = {
+            let cols = self.size_info.columns();
+            let lines = self.size_info.screen_lines();
+            Vec::with_capacity(cols * lines)
+        };
```

Comment:
This seems like an optimization unrelated to this PR? I've benchmarked this particular change myself previously and it did not have any impact on performance. Have you made any other observations or was this just an unrelated test you were performing?

Reply:
I just wrote that way, since it's always better not to do reallocs if you know upfront the size of allocation. It could show no difference for you, since your system if fast or your allocator is performing well, but you couldn't know how it'll work out with 'not smart enough' allocator or highly loaded system.

Reply:
> It could show no difference for you, since your system if fast or your allocator is performing well, but you couldn't know how it'll work out with 'not smart enough' allocator or highly loaded system.

That's a fair point, it's probably not going to be **worse**. Though this will definitely cause a bigger allocation than necessary since often the screen is likely not entirely full. Since this happens a bunch ideally the allocator won't have to ask the system anyway though.

Reply:
I really don't want to change this in this particular PR. I don't think it's uncontroversial enough to just slip it in.